src/annotations/items/AbstractAnnotationItem.h

```diff
@@ -52,6 +52,7 @@ class AbstractAnnotationItem : public QGraphicsWidget
 	virtual void scale(qreal sx, qreal sy) = 0;
 	virtual Tools toolType() const = 0;
 	virtual void setProperties(const PropertiesPtr &properties);
+	virtual bool allowsApplyingImageEffects() const;
```

Comment:
Instead of a bool that needs to be checked externally I think it would be better to have a method `applyImageEffect(ImageEffect *effect)` that mostly (any by default) does nothing but only the dupplicateItem applies it actually.